tests/loader/flash/test_flash_loader.py

```diff
@@ -0,0 +1,224 @@
+    )
+    # Create expected paths
+    expected_raw_path = expected_path / "raw" / "hdf" / sub_dir
+    expected_processed_path = expected_path / "processed" / "parquet"
+
+    # Create a fake file system for testing
+    fs.create_dir(expected_raw_path)
+    fs.create_dir(expected_processed_path)
+
+    # Instance of class with correct config and call initialize_paths
+    fl = FlashLoader(config=config)
+    data_raw_dir, data_parquet_dir = fl.initialize_paths()
+
+    assert expected_raw_path == data_raw_dir[0]
+    assert expected_processed_path == data_parquet_dir
+
+
+def test_initialize_paths_filenotfound(config_file):
+    """
+    Test FileNotFoundError during the initialization of paths.
+    """
+    # Test the FileNotFoundError
+    config = config_file
+    del config["core"]["paths"]
+    config["core"]["beamtime_id"] = "11111111"
+    config["core"]["year"] = "2000"
+
+    # Instance of class with correct config and call initialize_paths
+    fl = FlashLoader(config=config)
+    with pytest.raises(FileNotFoundError):
+        _, _ = fl.initialize_paths()
+
+
+def test_invalid_channel_format(config_file):
+    """
+    Test ValueError for an invalid channel format.
+    """
+    config = config_file
+    config["dataframe"]["channels"]["dldPosX"]["format"] = "foo"
+
+    fl = FlashLoader(config=config)
+
+    with pytest.raises(ValueError):
+        fl.read_dataframe()
+
+
+def test_group_name_not_in_h5(config_file):
+    """
+    Test ValueError when the group_name for a channel does not exist in the H5 file.
+    """
+    config = config_file
+    config["dataframe"]["channels"]["dldPosX"]["group_name"] = "foo"
+    h5_path = "FLASH1_USER3_stream_2_run43878_file1_20230130T153807.1.h5"
+    fl = FlashLoader(config=config)
+
+    with pytest.raises(ValueError) as e:
+        fl.create_dataframe_per_file(config["core"]["paths"]["data_raw_dir"] + h5_path)
+
+    assert str(e.value.args[0]) == "The group_name for channel dldPosX does not exist."
+
+
+def test_buffer_schema_mismatch(config_file):
+    """
+    Test function to verify schema mismatch handling in the FlashLoader's 'read_dataframe' method.
+
+    The test validates the error handling mechanism when the available channels do not match the
+    schema of the existing parquet files.
+
+    Test Steps:
+    - Attempt to read a dataframe after adding a new channel 'gmdTunnel2' to the configuration.
+    - Check for an expected error related to the mismatch between available channels and schema.
+    - Force recreation of dataframe with the added channel, ensuring successful dataframe creation.
+    - Simulate a missing channel scenario by removing 'gmdTunnel2' from the configuration.
+    - Check for an error indicating a missing channel in the configuration.
+    - Clean up created buffer files after the test.
+    """
+    fl = FlashLoader(config=config_file)
+
+    # Read a dataframe for a specific run
+    fl.read_dataframe(runs=["43878"])
```

Comment:
This creates again a race condition with at least one other test in the reader test using the same generic paths in the config. Maybe we should overwrite them everywhere, if actual parquet files are being created

Reply:
It's in test_timed_dataframe

Reply:
I will add the remove buffer at end of test_timed_dataframe
but I think we can think about introducing fixtures (or fixture factories I think) that allow for setting up test env and then a teardown (finish step)